File: src/extras-attica/libversion.cpp
#include <version.h>
#include "libversion.h"
#include "libversion.hxx"

unsigned int Attica_VersionMajor() {
    return Attica::versionMajor();
}

unsigned int Attica_VersionMinor() {
    return Attica::versionMinor();
}

unsigned int Attica_VersionRelease() {
    return Attica::versionRelease();
}

const char* Attica_VersionString() {
    return (const char*)Attica::versionString();
}

File: src_original/extras-attica/libversion.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");

/// https://api.kde.org/attica.html
pub const attica = struct {
    /// [Qt documentation](https://api.kde.org/attica.html#versionMajor)
    ///
    ///
    pub fn VersionMajor() u32 {
        return qtc.Attica_VersionMajor();
    }

    /// [Qt documentation](https://api.kde.org/attica.html#versionMinor)
    ///
    ///
    pub fn VersionMinor() u32 {
        return qtc.Attica_VersionMinor();
    }

    /// [Qt documentation](https://api.kde.org/attica.html#versionRelease)
    ///
    ///
    pub fn VersionRelease() u32 {
        return qtc.Attica_VersionRelease();
    }

    /// [Qt documentation](https://api.kde.org/attica.html#versionString)
    ///
    ///
    pub fn VersionString() []const u8 {
        const _ret = qtc.Attica_VersionString();
        return std.mem.span(_ret);
    }
};
